Docs 2.0 content technical review (#676)
graknlabs/docs: 0001-typedb/01-start/02-installation.md

Reviewer: @jamesreprise : Please perform technical review.

longTailKeywords: typedb on linux, typedb on mac, typedb on windows, start typedb server
summary: Install and run the TypeDB Server on Linux, Mac or Windows.
toc: false
---

# Installation

<!--- Reimplement tabs when they are product ready -->

## Docker

### Install

1. Ensure [Docker](https://docs.docker.com/get-docker/) is installed and running:

   <!-- test-ignore -->
      ```bash
      docker version
      ```

2. Pull the latest TypeDB Docker image:

   <!-- test-ignore -->
      ```bash
      docker pull vaticle/typedb:latest
      ```

### Start

The TypeDB container exposes TypeDB’s default port, `1729`, and uses its default data directory, 
`/opt/typedb-all-linux/server/data`.

The following command starts a named TypeDB container, maps its exposed port to the host machine and creates a named 
volume mapped to the data directory — making it persistent across container restarts.

<!-- test-ignore -->
```bash
docker run --name try-typedb -d -v try-typedb-data:/opt/typedb-all-linux/server/data/ -p 1729:1729 --platform linux/amd64 vaticle/typedb:latest
```

<div class="note">
[Note]

> The `--platform linux/amd64` parameter is required to run the TypeDB container on MacOS with an ARM64 architecture 

Reviewer 2: Could also call it ARM64/AArch64, as AArch64 is Arm's official name for the architecture.
```suggestion
The `--platform linux/amd64` parameter is required to run the TypeDB container on MacOS with an ARM64/AArch64 architecture. 
```

> The `--platform linux/amd64` parameter is required to run the TypeDB container on MacOS with an ARM64 architecture 

Reviewer 2: We also want to be careful to not have people think that Docker is our recommendation for macOS users - let's specify that in this note as well.

Author: @shane-k-j specifically asked me to put docker first as it is our default recommended option.

> (e.g. M1 processor).

Reviewer 2: References to specific processors will likely go out of date quickly, so let's remove this. 
```suggestion
```

   ```

2. Update the package cache:

   <!-- test-ignore -->
   ```bash
   sudo apt update
   ```
3. Ensure Java 11+ is installed:

   <!-- test-ignore -->
   ```bash
   sudo apt install openjdk-11-jre
   ```

   TypeDB supports the [OpenJDK](https://jdk.java.net) and
   [Oracle JDK](https://www.oracle.com/java/technologies/downloads/).

4. Check the latest version number for typedb-server and its dependencies:

   <!-- test-ignore -->
   ```bash
   apt show typedb-server
   ```

   This will show you console output like this:

   <!-- test-ignore -->
   ```bash
   Package: typedb-server
   Version: 2.15.0
   Priority: optional
   Section: contrib/devel
   Maintainer: Vaticle <community@vaticle.com>
   Installed-Size: unknown
   Depends: openjdk-11-jre, typedb-bin (=2.12.0)
   Download-Size: 71.8 MB
   APT-Sources: https://repo.vaticle.com/repository/apt trusty/main all Packages
   Description: TypeDB (server)
   ```

   Take a note of the latest typedb-server version shown at the `Package` field and the correspondent `typedb-bin` 
   package version shown at the `Depends` field.

>    To get some other version of TypeDB installed, we can use the `apt show typedb-server=2.14.1`, where `2.14.1` — 

Reviewer 2: Let's try and stick with a single version that users ought to substitute with their own to avoid confusion.
```suggestion
   To get some other version of TypeDB installed, we can use the `apt show typedb-server=2.15.0`, where `2.15.0` — 
```

Author: The key here is the word `other`. That's why we have another version number (not the default 2.15).

   the version we need.

5. Install `typedb-server`, `typedb-console`, and `typedb-bin` packages using the versions from the previous command 

>    (server and console have the same version number. Here we will use `2.15` for example):

Reviewer 2: ```suggestion
   (server and console have the same version number. Here we will use `2.15.0` for example):
```
